src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -270,6 +270,27 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             // Note: persist "m_currentThread" to be compatible with old versions
             info.AddValue(ThreadIDField, 0);
         }
+        
+        [System.Diagnostics.Conditional("DEBUG")]
+        private void DebugDump()
+        {
+            if (Length == 0)
+            {
+                return;
+            }
+            StringBuilder chunk = this;
+            // showing at most the last 10 chunks in this StringBuilder
+            const int maxChunksToPrint = 10;
+            var debugText = new StringBuilder();
+            for (int i = 0; chunk != null && i < maxChunksToPrint; i++)
+            {
+                debugText.Insert(0, "|" + new string(chunk.m_ChunkChars));
+                chunk = chunk.m_ChunkPrevious;
+            }
+            debugText.Append("|");
+
+            Debug.WriteLine(debugText.ToString().Replace("\0", "."));
```

Comment:
As discussed instead of writing to Debug stream, I suggest `yield return stringRepresentingAChunk;`

Reply:
Once you're returning an enumreable, you won't need the `|` to separate blocks. Why not add a second debug method to give a single string with `|` separators, simply:

```c#
public string DebugDumpString()
{
     return string.Join("|", DebugDump());
}
```